src/redismodule.h

```diff
@@ -733,6 +733,9 @@ REDISMODULE_API int (*RedisModule_GetKeyspaceNotificationFlagsAll)() REDISMODULE
 REDISMODULE_API int (*RedisModule_IsSubEventSupported)(RedisModuleEvent event, uint64_t subevent) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_GetServerVersion)() REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_GetTypeMethodVersion)() REDISMODULE_ATTR;
+REDISMODULE_API int (*RedisModule_AddCommandKeySpecRange)(RedisModuleCtx *ctx, const char *name, const char *specflags, int firstkey, int lastkey, int keystep) REDISMODULE_ATTR;
+REDISMODULE_API int (*RedisModule_AddCommandKeySpecKeyword)(RedisModuleCtx *ctx, const char *name, const char *specflags, const char *keyword, int keycount, int keystep) REDISMODULE_ATTR;
+REDISMODULE_API int (*RedisModule_AddCommandKeySpecKeynum)(RedisModuleCtx *ctx, const char *name, const char *specflags, int keynumidx, int firstkey, int keystep) REDISMODULE_ATTR;
```

Comment:
Don't like this, would be much happier with `RM_CreateCommandEx` or something but I acknowledge that modules will probably end up using both anyway which is ugly as well. Can't think of anything better, but please comment if you do!

Reply:
`RM_CreateCommandEx` would mean it needs to get a pointer to an array of structs, and we'll have to declare an ABI stable strut.

but the bigger problem is that modules that want to work with both redis versions (old and new) need to call the old RM_CreateCommand, and then call the new RM_AddCommandKeySpec variants optionally (only if they exist).

i.e. if we have a new RM_CreateCommandEx, modules will need to have an if-else and bother to call both.

Reply:
@oranagra yes, that's why I mentioned it's not necessarily better.